numpy/_core/tests/test_nditer.py

```diff
@@ -3379,6 +3378,40 @@ def test_partial_iteration_error(in_dtype, buf_dtype):
     assert count == sys.getrefcount(value)
 
 
+def test_arbitrary_number_of_ops():
+    # 2*16 + 1 is still just a few kiB, so should be fast an easy to deal with
+    # but larger than any small custom integer.
+    ops = [np.arange(10) for a in range(2**16 + 1)]
+
+    it = np.nditer(ops)
+    for i, vals in enumerate(it):
+        assert all(v == i for v in vals)
+
+
+def test_arbitrary_number_of_ops_nested():
+    # 2*16 + 1 is still just a few kiB, so should be fast an easy to deal with
+    # but larger than any small custom integer.
+    ops = [np.arange(10) for a in range(2**16 + 1)]
+
+    it = np.nested_iters(ops, [[0], []])
+    for i, vals in enumerate(it):
+        assert all(v == i for v in vals)
+
+
+@pytest.mark.slow
+@requires_memory(9 * np.iinfo(np.intc).max)
```

Comment:
shouldn't this be 8?  Or is there a byte for write flags or so? (not that it matters!)